### PR TITLE
Update binaryen

### PR DESCRIPTION
**Pull request by MaxGraey**
- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -258,25 +258,25 @@ export const enum ExpressionId {
...
-  StringConcat = 74 /* _BinaryenStringConcatId */,
-  StringEq = 75 /* _BinaryenStringEqId */,
-  StringAs = 76 /* _BinaryenStringAsId */,
-  StringWTF8Advance = 77 /* _BinaryenStringWTF8AdvanceId */,
-  StringWTF16Get = 78 /* _BinaryenStringWTF16GetId */,
-  StringIterNext = 79 /* _BinaryenStringIterNextId */,
-  StringIterMove = 80 /* _BinaryenStringIterMoveId */,
-  StringSliceWTF = 81 /* _BinaryenStringSliceWTFId */,
-  StringSliceIter = 82 /* _BinaryenStringSliceIterId */
+  ArrayInit = 65 /* _BinaryenArrayInitId */,
```
Comment (dcodeIO) [inline src/module.ts:261]:
```suggestion
  ArrayNewSeg = 64 /* TODO_BinaryenArrayNewSegId */,
  ArrayInit = 65 /* _BinaryenArrayInitId */,
```

File: tests/compiler/ternary.release.wat [hunk 2]
```diff
@@ -10,6 +10,7 @@
  (data (i32.const 1036) "\1c")
  (data (i32.const 1048) "\02\00\00\00\02\00\00\00a")
  (table $0 1 1 funcref)
+ (elem $0 (i32.const 1))
```
Comment (dcodeIO) [inline tests/compiler/ternary.release.wat:13]: Any idea what's going on here? Seems strange that this affects just a few release tests.

Comment (MaxGraey) [reply tests/compiler/ternary.release.wat:13]: No idea. Perhaps a some regression in Binaryen?

Comment (MaxGraey) [reply tests/compiler/ternary.release.wat:13]: debug fixture contain this: https://github.com/AssemblyScript/assemblyscript/blob/c7eafd336d843bc6bd2a0796613f731955cd8f63/tests/compiler/std/array-access.debug.wat#L23

**Comment by dcodeIO** [comment]
Would it be good to make the pass pipeline change still? :)

**Pull request closed without merging**